packages/rocketchat-channel-settings/client/views/channelSettings.js

```diff
@@ -164,7 +165,10 @@ Template.channelSettings.onCreated(function() {
 			save(value, room) {
 				let nameValidation;
 				if (!RocketChat.authz.hasAllPermission('edit-room', room._id) || (room.t !== 'c' && room.t !== 'p')) {
```

Comment:
What if we treat all of the default room types as the other "custom" types and provide sane defaults for them? That way code like this can be merged into one line and not be nested?

Reply:
@graywolf336 I thought of that as well. I wanted to keep the footprint of this PR as small as possible, 100% compatible - and make the review easier.

Reply:
@mrsimpson in my opinion, I would rather take the time to review the pull request now than spend time in the future following the path of the code each time it has to be interacted with...if that makes sense.

Reply:
@graywolf336 This absolutely makes sense! I would propose to modify all places as I started it. Then, in one "final" commit, I would implement the defined methods for the inbuilt room-types. wdyt?

Reply:
@mrsimpson Whichever way is easier for you will work for me. :+1:

Reply:
@graywolf336 I adapted the hard-coded values with interface calls. Can you please verify the merge-ability?